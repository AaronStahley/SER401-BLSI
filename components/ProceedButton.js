import React from 'react';
import {Button} from 'react-native';
import Colors from '../constants/Colors';

const { width, height} = Dimensions.get('window');

export default class ProceedButton extends React.Component {
   
    constructor(props) {
        super(props);
        this.state = {render: true};
    }

    render() {
        width = Dimensions.get("window").width;
        height = Dimensions.get("window").height;

        if(this.state.render) {
            return(<Button onPress={() => {this.setState({render: false});}}
                color={Colors.proceedButton} 
                buttonStyle={styles.proceedButton} 
                title={this.props.title} />);
        }
        else {
            return(null);
        }
    }
}