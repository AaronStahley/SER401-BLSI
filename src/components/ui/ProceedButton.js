import React from 'react';
import {Button, View, TouchableOpacity, Text, StyleSheet} from 'react-native';
import Colors from '../../common/Colors';

export default class ProceedButton extends React.Component {

    render() {
        return (

            <View style={styles.buttonContainer}>
                <TouchableOpacity style={styles.button} onPress={this.props.onPress}>
                    <Text style={styles.buttonText}>Proceed</Text>
                </TouchableOpacity>
            </View>

        );
    }
}

const styles = StyleSheet.create({
    proceedButton: {
        alignContent: 'center',
        marginTop   : 10,
        marginBottom: 10,
        borderRadius: 0
    },
    button: {
        flex: 1,
        backgroundColor: '#ee3e41',
        borderWidth: 0,
        borderRadius: 5,
        alignItems: 'center',
        margin: 5
    },
    buttonText:{
        color: '#fff',
        fontSize: 16,
        margin: 5
    },
    buttonContainer:{
        flexDirection: 'row',

    }
});