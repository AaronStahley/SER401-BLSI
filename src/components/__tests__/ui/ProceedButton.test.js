import 'react-native';
import React from 'react';
import ProceedButton from '../../ui/ProceedButton';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer
        .create(<ProceedButton title="snapshot" onPress={() => {}}></ProceedButton>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});