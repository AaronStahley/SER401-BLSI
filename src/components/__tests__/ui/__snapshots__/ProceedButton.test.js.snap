// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType="button"
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
          Object {
            "color": "#ee3e41",
          },
        ]
      }
    >
      snapshot
    </Text>
  </View>
</View>
`;
